File: jetbrains-plugin/src/main/kotlin/com/claudecodeplus/plugin/services/NotificationService.kt
package com.claudecodeplus.plugin.services

import com.intellij.notification.NotificationGroupManager
import com.intellij.notification.NotificationType
import com.intellij.openapi.project.Project

/**
 * 通知服务
 * 
 * 使用 IntelliJ Notification API 处理后台任务通知
 */
class NotificationService(private val project: Project) {
    
    companion object {
        private const val NOTIFICATION_GROUP_ID = "Claude Code Plus Notifications"
    }
    
    /**
     * 显示信息通知
     */
    fun showInfo(title: String, content: String = "") {
        NotificationGroupManager.getInstance()
            .getNotificationGroup(NOTIFICATION_GROUP_ID)
            .createNotification(title, content, NotificationType.INFORMATION)
            .notify(project)
    }
    
    /**
     * 显示警告通知
     */
    fun showWarning(title: String, content: String = "") {
        NotificationGroupManager.getInstance()
            .getNotificationGroup(NOTIFICATION_GROUP_ID)
            .createNotification(title, content, NotificationType.WARNING)
            .notify(project)
    }
    
    /**
     * 显示错误通知
     */
    fun showError(title: String, content: String = "") {
        NotificationGroupManager.getInstance()
            .getNotificationGroup(NOTIFICATION_GROUP_ID)
            .createNotification(title, content, NotificationType.ERROR)
            .notify(project)
    }
    
    /**
     * 显示工具调用需要批准的通知
     */
    fun notifyToolApprovalNeeded(toolName: String, onApprove: () -> Unit, onReject: () -> Unit) {
        val notification = NotificationGroupManager.getInstance()
            .getNotificationGroup(NOTIFICATION_GROUP_ID)
            .createNotification(
                "工具调用需要批准",
                "Claude 请求执行工具: $toolName",
                NotificationType.INFORMATION
            )
        
        notification.addAction(object : com.intellij.openapi.actionSystem.AnAction("批准") {
            override fun actionPerformed(e: com.intellij.openapi.actionSystem.AnActionEvent) {
                onApprove()
                notification.expire()
            }
        })
        
        notification.addAction(object : com.intellij.openapi.actionSystem.AnAction("拒绝") {
            override fun actionPerformed(e: com.intellij.openapi.actionSystem.AnActionEvent) {
                onReject()
                notification.expire()
            }
        })
        
        notification.notify(project)
    }
    
    /**
     * 显示文件编辑成功通知
     */
    fun notifyFileEdited(filePath: String) {
        showInfo("文件已编辑", filePath)
    }
    
    /**
     * 显示后台任务完成通知
     */
    fun notifyTaskCompleted(taskName: String) {
        showInfo("任务完成", taskName)
    }
    
    /**
     * 显示后台任务失败通知
     */
    fun notifyTaskFailed(taskName: String, error: String) {
        showError("任务失败", "$taskName: $error")
    }
}

File: vscode-extension/src/ide/services/notificationService.ts
/**
 * Notification Service
 * 
 * Uses VS Code Notification API for background task notifications
 * Translated from jetbrains-plugin/src/main/kotlin/com/asakii/plugin/services/NotificationService.kt
 */

import * as vscode from 'vscode';

/**
 * Notification Service
 * 
 * Handles notifications for Claude Code Plus
 */
export class NotificationService {
    private static instance: NotificationService | undefined;
    
    private constructor() {}
    
    static getInstance(): NotificationService {
        if (!this.instance) {
            this.instance = new NotificationService();
        }
        return this.instance;
    }
    
    /**
     * Show information notification
     */
    showInfo(title: string, content?: string): void {
        const message = content ? `${title}: ${content}` : title;
        vscode.window.showInformationMessage(message);
    }
    
    /**
     * Show warning notification
     */
    showWarning(title: string, content?: string): void {
        const message = content ? `${title}: ${content}` : title;
        vscode.window.showWarningMessage(message);
    }
    
    /**
     * Show error notification
     */
    showError(title: string, content?: string): void {
        const message = content ? `${title}: ${content}` : title;
        vscode.window.showErrorMessage(message);
    }
    
    /**
     * Show tool approval notification
     */
    async notifyToolApprovalNeeded(
        toolName: string,
        onApprove: () => void,
        onReject: () => void
    ): Promise<void> {
        const result = await vscode.window.showInformationMessage(
            `Claude requests to execute tool: ${toolName}`,
            'Approve',
            'Reject'
        );
        
        if (result === 'Approve') {
            onApprove();
        } else if (result === 'Reject') {
            onReject();
        }
    }
    
    /**
     * Show file edited notification
     */
    notifyFileEdited(filePath: string): void {
        this.showInfo('File Edited', filePath);
    }
    
    /**
     * Show task completed notification
     */
    notifyTaskCompleted(taskName: string): void {
        this.showInfo('Task Completed', taskName);
    }
    
    /**
     * Show task failed notification
     */
    notifyTaskFailed(taskName: string, error: string): void {
        this.showError('Task Failed', `${taskName}: ${error}`);
    }
    
    /**
     * Show progress notification with cancel option
     */
    async withProgress<T>(
        title: string,
        task: (
            progress: vscode.Progress<{ message?: string; increment?: number }>,
            token: vscode.CancellationToken
        ) => Promise<T>
    ): Promise<T> {
        return vscode.window.withProgress(
            {
                location: vscode.ProgressLocation.Notification,
                title,
                cancellable: true
            },
            task
        );
    }
    
    /**
     * Show status bar message
     */
    showStatusBarMessage(message: string, timeoutMs: number = 3000): vscode.Disposable {
        return vscode.window.setStatusBarMessage(message, timeoutMs);
    }
}

// Export singleton instance
export const notificationService = NotificationService.getInstance();
